server/py/services/api/crud/workflows.py

```diff
@@ -627,3 +636,91 @@ def get_workflow_id(
+        run_uid: str,
+        rerun_request: mlrun.common.schemas.RerunWorkflowRequest,
+        auth_info: mlrun.common.schemas.AuthInfo = None,
+    ) -> mlrun_model.RunObject:
+        """
+        Run a rerun workflow runner.
+
+        :param runner:         Workflow runner function object.
+        :param project:        MLRun project.
+        :param run_uid:        UID of the original failed run to retry.
+        :param rerun_request:  RerunWorkflowRequest containing any notifications and retry parameters.
+        :param auth_info:      Authentication information of the request.
+        :return: RunObject for the rerun.
+        """
+
+        labels = {
+            mlrun_constants.MLRunInternalLabels.project: project.metadata.name,
+            mlrun_constants.MLRunInternalLabels.job_type: JOB_TYPE_RERUN_WORKFLOW_RUNNER,
+            mlrun_constants.MLRunInternalLabels.workflow: runner.metadata.name,
+            mlrun_constants.MLRunInternalLabels.original_workflow_id: run_uid,
+        }
+
+        self._enrich_runner_node_selector(runner, rerun_request)
+
+        return self.prepare_and_run(
+            runner=runner,
+            project=project,
+            labels=labels,
+            auth_info=auth_info,
+            artifact_path=mlrun_config.config.artifact_path,
+            rerun_request=rerun_request,
+        )
+
+    def _prepare_run_object(
+        self,
+        project: mlrun.common.schemas.ProjectOut,
+        labels: dict[str, str],
+        rerun_request: mlrun.common.schemas.RerunWorkflowRequest,
+        workflow_request: Optional[mlrun.common.schemas.WorkflowRequest] = None,
+        run_name: Optional[str] = None,
+        uid: Optional[str] = None,
+        scrape_metrics: Optional[str] = None,
+        url: str = "",
+    ) -> mlrun_model.RunObject:
+        """
+        Prepare the RunObject for rerunning the workflow.
+
+        :param project:          MLRun project.
+        :param labels:           Labels for the run.
+        :param workflow_request: Workflow request containing the workflow spec.
+        :param run_name:         Name of the run.
+        :param uid:              Unique identifier for the run.
+        :param scrape_metrics:   Whether to scrape metrics.
+        :return: RunObject ready for execution.
+        """
+
+        notifications = [
+            mlrun_model.Notification.from_dict(notification.dict())
+            for notification in rerun_request.notifications or []
+        ]
+
+        run_object = self._create_run_object(
+            source=project.spec.source,
+            project_name=project.metadata.name,
+            save=False,
+            handler="mlrun.projects.rerun_workflow",
+            parameters={
+                "run_uid": rerun_request.rerun_id,
+                "project_name": project.metadata.name,
+            },
+            notifications=notifications,
+            run_name=f"rerun-{rerun_request.run_name}-{rerun_request.rerun_id}",
```

Comment:
could it be too long? I suggest to remote the rerun id from the name - adding perhaps an increment of the reruns could be an option?